src/Widgets/ToolbarButton.vala

```diff
@@ -77,16 +77,27 @@ public class ENotes.ToolbarButton : Gtk.Button {
         clicked.connect (() => {
             Gtk.TextIter start, end;
             code_buffer.get_selection_bounds (out start, out end);
-
+            
             if (code_buffer.has_selection) {
-                var text = start.get_text (end);
-
+                var text = start.get_text (end).strip ();
```

Comment:
Since you're doing a `.strip ()` here, just make sure you're not accidentally putting the text together with your selection :)

So, if you had the words `hello world` and you selected hello and the space, trying to make the text bold should move the `**` right up to the hello word, but leave the space in between. 

Hint: You might have to store the removed spaces somewhere ;)

Reply:
Nice catch, that was actually a last minute change due to the thought that whitespaces can't be bolded, but you're right. Will review it ;)

Reply:
You're actually correct that whitespaces can't be bolded, and actually the syntax highlight doesn't like if you you do it :sweat_smile: 

So if you could fix that it would be awesome! If not then that can always land in a separate PR